File: Python/max-consecutive-ones.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result, local_max = 0, 0
        for n in nums:
            local_max = (local_max + 1 if n else 0)
            result = max(result, local_max)
        return result

File: C++/max-consecutive-ones.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int result = 0, local_max = 0;
        for (const auto& n : nums) {
            local_max = n ? local_max + 1 : 0;
            result = max(result, local_max);
        }
        return result;
    }
};
